src/Marten/Linq/Parsing/LinqHandlerBuilder.cs

```diff
@@ -268,10 +268,9 @@ private void wrapIncludes(IList<IIncludePlan> includes)
             }
 
             var statement = new IncludeIdentitySelectorStatement(TopStatement, includes, _session);
+            TopStatement.Limit = TopStatement.Offset = 0;
```

Comment:
Le fait de bouger ici, ca place le offset.limit a quel endroit dans la requete SQL?

Reply:
Lorsqu'il y a des `include` la requete est fais en 2 temps dans la db
- select ids left join included relationships
- select * from the relevant tables where ids are in the first query

Initialement le paging est seulement appliquer sur la 2eme query ce qui fait que plus on a de resources dans la db, plus la query est longue meme si on page, puisque que le left join initial est sur toute la table.

J'ai bouger le paging + where statement dans le statement initial pour s'assurer que quand on a du paging, ben les requete prenne pas 3 secondes a rouler meme si on retourne juste 20 resultats.

Reply:
Je comprend bien, va falloir faire attention parce que ca peut devenir prone a des comportements non expected. On peut perdre le "contexte" de la requete.

Reply:
Je suis d'accord avec toi, les changements que j'ai fais semblent relativement ciblé mais ca a défait les filtres de tenancy à un moment donnée et j'ai du ajuster.